crates/ide-assists/src/handlers/convert_iter_for_each_to_for.rs

```diff
@@ -218,6 +221,19 @@ fn validate_method_call_expr(
     let iter_trait = FamousDefs(sema, krate).core_iter_Iterator()?;
     it_type.impls_trait(sema.db, iter_trait, &[]).then(|| (expr, receiver))
 }
+pub(crate) fn detect_for_loop(
```

Comment:
This should probably be moved to some utils module, I don't think we should cross-import stuff between assist modules like that

Reply:
This function is from first few lines of "convert_for_loop_with_for_each". I'm not sure if I should copy it or put it into some function (here is detect_for_loop) and move it in utils.